chore: handover to governance

LHerskind opened the pull request:
Adds a test to se if we are correctly updating the ownerships and handings things over to governance. 

Updated part of the deployment setup to properly hand over things. Fixed an issue with return values as we did not actually return the coinissuer but just whoever owned the fee asset - which were not the same because it was not handed over.

LHerskind commented:
* **#16630** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/16630?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/16630?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `next`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

LHerskind commented:
Had to deal with a bunch of tests that abused their rights over the asset + strange failure where the caller just did not have any eth to pay to update the proving cost per mana.

LHerskind commented on l1-contracts/src/mock/TestERC20.sol:
```diff
@@ -34,11 +34,8 @@ contract TestERC20 is ERC20, IMintableERC20, Ownable2Step {
   }
 
   function acceptOwnership() public virtual override(Ownable2Step) {
-    address oldOwner = owner();
```
Keep the old owner as minter in order to not go completely mad on the tests.

LHerskind commented on yarn-project/constants/src/constants.ts:
```diff
@@ -1,5 +1,4 @@
 // Typescript-land-only constants
-export const FEE_FUNDING_FOR_TESTER_ACCOUNT = BigInt(1_000e18);
```
Deleted as we are should not specify the amount it should just be same as the fee asset handler will do. And we were abusing it in plenty tests without it actually being for fees.

LHerskind commented on yarn-project/end-to-end/src/e2e_epochs/epochs_test.ts:
```diff
@@ -92,7 +92,7 @@ export class EpochsTestContext {
     const ethereumSlotDuration = opts.ethereumSlotDuration ?? envEthereumSlotDuration;
     const aztecSlotDuration = opts.aztecSlotDuration ?? ethereumSlotDuration * 2;
-    const aztecEpochDuration = opts.aztecEpochDuration ?? 4;
+    const aztecEpochDuration = opts.aztecEpochDuration ?? 6;
```
Since the setups are now more txs as it is doing more stuff, the epoch duration should go up to not break a slew of tests.

LHerskind commented on yarn-project/end-to-end/src/e2e_epochs/epochs_upload_failed_proof.test.ts:
```diff
@@ -73,7 +73,7 @@ describe('e2e_epochs/epochs_upload_failed_proof', () => {
     // and wait for the data to be uploaded to the remote file store
     await test.waitUntilEpochStarts(1);
-    await retryUntil(() => epochUploadUrl !== undefined, 'Upload epoch failure', 120, 1);
+    await retryUntil(() => epochUploadUrl !== undefined, 'Upload epoch failure', 240, 1);
```
With the epoch size being bigger, this task takes longer.

LHerskind commented on yarn-project/end-to-end/src/fixtures/snapshot_manager.ts:
```diff
@@ -379,31 +377,6 @@ async function setupFromFresh(
   aztecNodeConfig.rollupVersion = deployL1ContractsValues.rollupVersion;
   aztecNodeConfig.l1PublishRetryIntervalMS = 100;
 
-  if (opts.fundRewardDistributor) {
```
This one did not really make sense to me, as we were already funding the reward distributor as part of the deployment.

LHerskind commented on yarn-project/ethereum/src/deploy_l1_contracts.test.ts:
```diff
@@ -159,4 +162,99 @@ describe('deploy_l1_contracts', () => {
     );
   });
+
+  it('ensure governance is the owner', async () => {
```
This test is really what brought all the other changes. Ensuring that governance have actually gotton ownership of things 👍

LHerskind commented on yarn-project/ethereum/src/deploy_l1_contracts.ts:
```diff
@@ -246,36 +246,14 @@ export const deploySharedContracts = async (
-    1n * 10n ** 18n, // @todo  #8084
-    governanceAddress.toString(),
+    1_000_000n * 10n ** 18n, // @todo  #8084
+    l1Client.account.address,
```
Initially we will have the ownership of the coinissuer, since we need the coinissuer to become the owner of the fee asset, which is a two-step ownable.

LHerskind commented on yarn-project/ethereum/src/deploy_l1_contracts.ts:
```diff
@@ -745,6 +760,54 @@ export const handoverToGovernance = async (
     txHashes.push(transferOwnershipTxHash);
   }
 
+  if (acceleratedTestDeployments || (await feeAsset.read.owner()) !== coinIssuerAddress.toString()) {
```
First handover the fee asset.

LHerskind commented on yarn-project/ethereum/src/deploy_l1_contracts.ts:
```diff
@@ -745,6 +760,54 @@ export const handoverToGovernance = async (
+  }
+
+  // If the owner is not the Governance contract, transfer ownership to the Governance contract
+  if (
```
Then the coinissuer

LHerskind commented on aztec-up/test/bridge_and_claim.sh:
```diff
@@ -42,7 +42,7 @@ aztec-wallet \
 
 aztec-wallet \
-  bridge-fee-juice 1000000000000000000 accounts:main \
+  bridge-fee-juice 1000000000000000000000 accounts:main \
```
The mint amount was changed. I'm not really sure why we actually pass this amount in for the mint case, as we won't allow anything but the correct mint amount 🤷

just-mitch replied:
One day https://github.com/AztecProtocol/aztec-packages/pull/15541 will land...

just-mitch replied:
Created https://linear.app/aztec-labs/issue/TMNT-241/clean-up-bridging to track getting #15541 merged, slated for the upcoming cycle.

just-mitch commented on yarn-project/aztec-faucet/terraform/main.tf:
```diff
@@ -148,7 +148,7 @@ resource "aws_ecs_task_definition" "aztec-faucet" {
         {
           name  = "EXTRA_ASSET_AMOUNT",
-          value = "1000000000000000000"
+          value = "1000000000000000000000"
```
I think this entire faucet is dead code.

just-mitch replied:
Created https://linear.app/aztec-labs/issue/TMNT-242/audit-infrastructure-and-code

just-mitch commented on yarn-project/end-to-end/src/e2e_epochs/epochs_proof_fails.test.ts:
```diff
@@ -85,16 +95,23 @@ describe('e2e_epochs/epochs_proof_fails', () => {
       const prover = originalCreate();
       jest.spyOn(prover, 'finalizeEpoch').mockImplementation(async () => {
-        const seconds = L2_SLOT_DURATION_IN_S * test.epochDuration;
+        const seconds = L2_SLOT_DURATION_IN_S * (test.epochDuration * 2); // Forgive me for I have sinned.
```
I have a theory that every problem in distributed systems can be "solved trivially" by adding a delay 😏

just-mitch commented on yarn-project/end-to-end/src/e2e_fees/bridging_race.notest.ts:
Why is this file `.notest.ts`?

LHerskind replied:
I have no clue.

just-mitch reviewed (approved):
NIce work battling the TS land 😤

AztecBot committed:
chore: handover to governance

Adds a test to se if we are correctly updating the ownerships and handings things over to governance.

Updated part of the deployment setup to properly hand over things. Fixed an issue with return values as we did not actually return the coinissuer but just whoever owned the fee asset - which were not the same because it was not handed over.

Co-authored-by: LHerskind <16536249+LHerskind@users.noreply.github.com>
